File: ios/Sources/CapacitorCalendarPlugin/Models/Results/CreateEventResult.swift
import Capacitor

struct CreateEventResult: JSResult {
    let id: String

    init(id: String?) throws {
        guard let id = id else {
            throw PluginError.failedToRetrieveEventId
        }
        self.id = id
    }

    func toJSON() -> JSObject {
        var result = JSObject()
        result["id"] = id
        return result
    }
}

File: android/src/main/java/dev/barooni/capacitor/calendar/models/results/CreateEventResult.kt
package dev.barooni.capacitor.calendar.models.results

import com.getcapacitor.JSObject
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.models.templates.JSResult

data class CreateEventResult(
    val id: Long?,
) : JSResult {
    init {
        if (id == null) {
            throw PluginError.FailedToRetrieveEventId
        }
    }

    override fun toJSON(): JSObject {
        val result = JSObject()
        result.put("id", id.toString())
        return result
    }
}
